pkg/neotest/signer.go

```diff
@@ -0,0 +1,101 @@
+package neotest
+
+import (
+	"bytes"
+
+	"github.com/nspcc-dev/neo-go/pkg/config/netmode"
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/hash"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+)
+
+// Signer is a generic interface which can be either simple- or multi-signature signer.
+type Signer interface {
+	// ScriptHash returns signer script hash.
+	Script() []byte
+	// Script returns signer verification script.
+	ScriptHash() util.Uint160
+	// SignHashable returns invocation script for signing an item.
+	SignHashable(uint32, hash.Hashable) []byte
+	// SignTx signs a transaction.
+	SignTx(netmode.Magic, *transaction.Transaction) error
+}
+
+// signer represents simple-signature signer.
+type signer wallet.Account
+
+// multiSigner represents single multi-signature signer consisting of provided accounts.
+type multiSigner []*wallet.Account
+
+// NewSingleSigner returns multi-signature signer for the provided account.
+// It must contain exactly as many accounts as needed to sign the script.
+func NewSingleSigner(acc *wallet.Account) Signer {
+	return (*signer)(acc)
+}
+
+// Script implements Signer interface.
+func (s *signer) Script() []byte {
+	return (*wallet.Account)(s).Contract.Script
```

Comment:
What if it's a standard signature signer (non-contract)?

Reply:
Added a check in constructor.